node/src/reactor/joiner.rs

```diff
@@ -593,11 +593,17 @@ impl reactor::Reactor for Reactor {
             .protocol_config
             .verifiable_chunked_hash_activation;
         let protocol_version = &chainspec_loader.chainspec().protocol_config.version;
+        let node_state = if config.node.archival_sync {
```

Comment:
I'm surprised this is still being called `archival_sync`. Should probably be renamed.

Reply:
Yes, it was cause I didn't merge the changes from the related PR, I merged the changes